js/form-validator.js

```diff
@@ -0,0 +1,103 @@
+import url from 'url';
```

Comment:
This is all existing code but I moved it to here (`/js/form-validator.js`). In a followup PR I'll remove `/pages/add/form/validator.js` and make the Add page uses this js file instead.